### PR TITLE
take marketplace as an argument from the frontend

### PR DESCRIPTION
**Pull request by dr-bonez**
## type changes:
`ServerInfo` (`/server-info`):
```diff
-   "status": "running" | "updated" | "updating" | "backing-up",
-   "update-progress": null | { "size": number | null, "downloaded": number },
+   "status-info": {
+       "backing-up": bool,
+       "updated": bool,
+       "update-progress": null | { "size": number | null, "downloaded": number },
+   }
```
`InstalledPackageDataEntry` (`/package-data/<id>/installed`):
```diff
+   "marketplace-url": string | null,
+   "developer-key": string,
```
## new rpc endpoints:
`marketplace.get`:
params: `{ "url": string }`
response: `any`

### CODE REVIEW

**Comment by dr-bonez** [comment]
Need to pre-download static files from the marketplace. Can't use /marketplace urls anymore

**Review by dr-bonez — COMMENTED** [review]
Need to pre-download static files from the marketplace. Can't use /marketplace urls anymore

File: backend/src/install/mod.rs [hunk 1]
```diff
@@ -489,6 +510,7 @@ pub async fn uninstall_impl(ctx: RpcContext, id: PackageId) -> Result<WithRevisi
 pub async fn download_install_s9pk(
     ctx: &RpcContext,
     temp_manifest: &Manifest,
+    marketplace_url: Option<Url>,
```
Comment (ProofOfKeags) [inline backend/src/install/mod.rs:513]: What is the meaning of `None`?

Comment (dr-bonez) [reply backend/src/install/mod.rs:513]: sideloaded

**Comment by dr-bonez** [comment]
alright this is ready to go

**Review by Blu-J — APPROVED** [review]

**Pull request merged**